python/cuml/cuml/tests/test_sklearn_compatibility.py

```diff
@@ -0,0 +1,103 @@
+        "check_estimators_unfitted": "LogisticRegression does not raise NotFittedError before fit",
+        "check_do_not_raise_errors_in_init_or_set_params": "LogisticRegression raises errors in init or set_params",
+        "check_n_features_in_after_fitting": "LogisticRegression does not check n_features_in consistency",
+        "check_sample_weights_not_an_array": "LogisticRegression does not handle non-array sample weights",
+        "check_sample_weights_list": "LogisticRegression does not handle list sample weights",
+        "check_sample_weight_equivalence_on_dense_data": "LogisticRegression sample weight equivalence not implemented",
+        "check_sample_weight_equivalence_on_sparse_data": "LogisticRegression does not handle sparse data",
+        "check_complex_data": "LogisticRegression does not handle complex data",
+        "check_dtype_object": "LogisticRegression does not handle object dtype",
+        "check_estimators_empty_data_messages": "LogisticRegression does not handle empty data",
+        "check_estimators_nan_inf": "LogisticRegression does not check for NaN and inf",
+        "check_estimator_sparse_tag": "LogisticRegression does not support sparse data",
+        "check_classifier_data_not_an_array": "LogisticRegression does not handle non-array data",
+        "check_classifiers_one_label": "LogisticRegression cannot train with one class",
+        "check_classifiers_train": "LogisticRegression does not handle list inputs",
+        "check_classifiers_train(readonly_memmap=True)": "LogisticRegression does not handle readonly memmap",
+        "check_classifiers_train(readonly_memmap=True,X_dtype=float32)": "LogisticRegression does not handle readonly memmap with float32",
+        "check_classifiers_regression_target": "LogisticRegression does not handle regression targets",
+        "check_supervised_y_no_nan": "LogisticRegression does not check for NaN in y",
+        "check_supervised_y_2d": "LogisticRegression does not handle 2D y",
+        "check_class_weight_classifiers": "LogisticRegression does not handle class weights properly",
+        "check_parameters_default_constructible": "LogisticRegression parameters are mutated on init",
+        "check_fit2d_1sample": "LogisticRegression does not handle single sample",
+        "check_set_params": "LogisticRegression does not handle set_params properly",
+        "check_fit1d": "LogisticRegression does not raise ValueError for 1D input",
+        "check_fit2d_predict1d": "LogisticRegression does not handle 1D prediction input gracefully",
+        "check_requires_y_none": "LogisticRegression does not handle y=None",
+    },
+}
+
+
+def get_xfails(estimator):
+    return PER_ESTIMATOR_XFAIL_CHECKS.get(type(estimator), {})
+
+
+def _check_sklearn_version():
+    """Check if scikit-learn version is >= 1.7"""
+    import sklearn
+    from packaging import version
+
+    return version.parse(sklearn.__version__) >= version.parse("1.7")
```

Comment:
Instead of doing any of this, you could add this towards the top of the file to skip all tests here if sklearn is < 1.7. Then you don't have to guard the definition of `test_sklearn_compatible_estimator`.

```
pytest.importorskip("sklearn", minversion="1.7")
```

Reply:
It raises an exception and aborts loading the rest of the file.

For other marks that you want to apply to every test in the file (but still load the file) you can also define 

```
pytestmark = ...
```

which may also be useful for conditional skipping/xfailing.